zyte_common_items/pipelines.py

```diff
@@ -61,6 +60,11 @@ def __init__(self):
         )
 
     def process_item(self, item, spider):
+        # As global level import can raise Deprecation warning for other classes
+        # of this same module which can be mis-leading. Hence moving this import here
+        # as the module is anyway going to deprecate and less likely used.
+        from . import ae
```

Comment:
Maybe it is best to move the import to `__init__` and set `self._downgrade` to `ae.downgrade`, so that here you can use `return self._downgrade(item)`.

Also, I would not bother with a comment for the reason here. Although a test that you can use the other middleware without getting a warning would be nice.

Reply:
Yes, that is another way possible. But in this case, at every new instantiation, module will be imported. As module is anyway not getting used extensively, we can definitely do it this way.
Thanks!